### DISCUSSION TITLE
Setting up SSL

### DISCUSSION THREAD

**Discussion by ZuluPilgrim** [pglombardo/PasswordPusher#952] [Q&A] [answered]
Ok, so I think I am now coming to the end of my questions. Is there a tutorial I can follow on setting up pwdpusher so that I only uses HTTPS connections? I am thinking that I should have probably done a crash course in docker first :)

**Comment by pglombardo** [comment, marked as answer]
Sure - To do that, there are two steps:

1. Get nginx running with Password Pusher as a backend.  I have an [example here](https://github.com/pglombardo/PasswordPusher/tree/master/containers/examples/pwpush-and-nginx) but there are many tutorials online about setting up nginx.
2. Learn how to generate and add SSL certificates to nginx.  This is beyond this Password Project but there are a ton of resources online to get you started.

So with this setup, users establish SSL connections to nginx and requests are proxied to Password Pusher.

Hope this helps!

**Reply by pglombardo** [reply]
@ZuluPilgrim Let me add that there have been a lot of nginx discussions [in the issues section of this repo](https://github.com/pglombardo/PasswordPusher/issues?q=is%3Aissue+nginx) that might be of help.
